### ISSUE TITLE
minting v3

### ISSUE THREAD

**Issue by xmonader** [threefoldtech/tfchain_pallets#48]
TODO

**Comment by LeeSmet** [comment]
This comment is not a technical description, but rather a summary of some things which need to be cleared out first.

## Provided formulas and data

It seems that the main data source to extract info from is the library, specifically https://library.threefold.me/info/threefold#/tfgrid/farming/threefold__farming_reward and the linked calculator sheet. An immediate problem is that both of these have different definitions of CU and SU. It is not immediately clear how much USD is used per  CU and SU in the wiki, but in the sheet 2.4USD and 1.5USD are used. The calculations in the wiki return a result which indicates that the wiki uses the same values.

The different CU and SU formulas:

### Wiki 

`CU = 2 vCPU, 4GB mem, 50 GB storage`. It is not clear if storage is HDD or SSD or both, but it's assumed to be SSD only. This then equates to `CU = MIN(CRU/2, MRU/4, SRU/50)`. Similarly, `SU = 1TB of net storage`. This is assumed to be both HDD and howerver much SSD is left (after subtracting the amount of SSD used for CU), which then equates to `SU = HRU/1000 + (SRU - (50*CU)) / 1000).

### Calculator

`CU = MIN((MRU-1)/4, CRU*4/2)`, which simplifies to `CU = MIN((MRU-1)/4, CRU*2)`
`SU = HRU/1200 + (SRU/300) * 0..8`, which simplifies to `SU = HRU/1200 + SRU/375`

### Comparison

Comparing the two, we see that the wiki requires storage space for Compute units whereas the calculator does not. Next, the wiki requires 2 vCPU per Cu, the calculator only needs 0.5 (so a vCPU is 4 times as valuable). Both require 4 GB memory per CU, though the calculator strips 1 GB (presumably reserved for the OS). The wiki states that 1 SU is 1TB of netto usable storage. There is no distinction between HDD or SSD here. Given that the price for SSD storage is outright higher than HDD storage, this would mean that it makes economically the most sense to only provide as much SSD storage as needed for the CU, and use HDD for everything else. That would lead to a very limited supply of SSD capacity on the grid (which in turn would make it practically unuseable).

The farming reward page (https://library.threefold.me/info/threefold#/tfgrid/farming/threefold__farming_hardware_overview?id=do-it-yourself) lists a recommendation of 1 vCpu to 4GB memory. It should be noted that neither of the above actually use this ratio. The wiki formula has a ratio of 1:2, whereas the formula in the calculator has a ration of 1:8.

The calculator mentions a reward for NU, this is not mentioned on the wiki page. This needs further clarification.

### V2 minting

The old minting does not pay the CU and SU, but rather the CPR. CPR is defined as `CPR = CU * 1.5 + SU`. A payout of 40/6 USD per CPR is given. This equates to a payout of `10USD/CU` and `6.67 USD/SU`. It should be noted that this was scaled down based on the total amount of tokens. For the v2 minting, the following calculations are used:

`CU = MIN(CRU*4, (MRU-1)/4)`
`SU = HRU/1200 + SRU/120`

This means that, SSD storage has significantly devaluated (as it equates to less SU in either case). Similarly, Cpu capacity has devaluated, as you need more cores in either case to result in the same amount of CU (assuming sufficient memory is available).

## What does this mean for the titan V2.1

The titan v2.1, currently listed on https://shop.threefold.tech, has [the following specifications](https://shop.threefold.tech/index.php?route=product/product&path=59&product_id=53):

4 core / 8 thread cpu,
32 GB memory
1TB nvme SSD

Furthermore, this presumably equates to 7.75CU and 4.17SU

Running the formulas extracted above:

wiki: 
CU = MIN(8/2, 32/4, 1000/50) = 4
SU = 0/1000 + (1000 - 50 * 4) / 1000 = 0.8

So according to the wiki formula, the values are  `CU = 4, SU = 0.8`

calculator:
CU = MIN((32-1)/4, 8*2) = 7.75
SU = 0/1200 + 1000/375 = 2.67

The calculator is closer to the listed info, but SU is still off. Final values: `CU = 7.75, SU = 2.67`

v2 minting:

CU = MIN(8*4, (32-1)/4) = 7.75
SU = 0/1200 + 1000/120 = 8.34

This results in the same amount of CU, but more SU than listed.

### Titan earnings

It seems that the reward will be 2.4USD/CU and 1.5USD/SU in either case. This leads to the following montly payout:

wiki: 4CU * 2.4USD/CU + 0.8SU * 1.5USD/SU = 10.8 USD * 1.25 -> 13.5 USD
calculator: 7.75CU * 2.4USD/CU + 2.67SU *1.5USD/SU =  22.605USD * 1.25 -> 28.25625 USD
store page: 7.75CU * 2.4USD/CU + 4.17SU * 1.5SU/USD = 24.855USD * 1.25 -> 31.06875 USD

In all cases, a 1.25 multiplier was added as the titan node is certified by default. Any earnings made from titan would come due to a serious increase in price _after_ the node was connected.

**Comment by andhartl** [comment]
According to Kristof the calculator formulas are the ones to go with. The SU formula assumes that an average of 20GB are used for Customer. Thats why it says 0.8 
<img width="593" alt="image" src="https://user-images.githubusercontent.com/18591016/141746799-e6ac15fd-a561-4e0e-a38a-89f18a3e0887.png">

**Comment by LeeSmet** [comment]
> The SU formula assumes that an average of 20GB are used for Customer.

What does this mean?

**Comment by andhartl** [comment]
we use the calculator formulas for minting. need to adjust the ones on the wiki. will take care of that.

**Comment by LeeSmet** [comment]
yes, I get that, I meant the specific sentence I linked. "20GB average used for Customer"?
